src/steps/object-field-equals.ts

```diff
@@ -0,0 +1,108 @@
+import { Field, ExpectedRecord } from './../core/base-step';
+/*tslint:disable:no-else-after-return*/
+
+import { BaseStep, StepInterface } from '../core/base-step';
+import { Step, RunStepResponse, FieldDefinition, StepDefinition, RecordDefinition } from '../proto/cog_pb';
+import * as util from '@run-crank/utilities';
+import { baseOperators } from '../client/constants/operators';
+
+export class ObjectFieldEquals extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Check a field on a Salesforce Object';
+  /* tslint:disable-next-line:max-line-length */
+  protected stepExpression: string = 'the (?<field>[a-zA-Z0-9_]+) field on salesforce (?<objName>[a-zA-Z0-9]+) object with id (?<id>[^\s]+) should (?<operator>be less than|be greater than|be|contain|not be|not contain) (?<expectedValue>.+)';
+  protected stepType: StepDefinition.Type = StepDefinition.Type.VALIDATION;
+  protected expectedFields: Field[] = [{
+    field: 'field',
+    type: FieldDefinition.Type.STRING,
+    description: 'Field name to check',
+  }, {
+    field: 'objName',
+    type: FieldDefinition.Type.STRING,
+    optionality: FieldDefinition.Optionality.OPTIONAL,
+    description: 'Salesforce object name',
+  }, {
+    field: 'id',
+    type: FieldDefinition.Type.STRING,
+    optionality: FieldDefinition.Optionality.OPTIONAL,
+    description: 'Object ID',
+  }, {
+    field: 'operator',
+    type: FieldDefinition.Type.STRING,
+    optionality: FieldDefinition.Optionality.OPTIONAL,
+    description: 'Check Logic (be, not be, contain, not contain, be greater than, or be less than)',
+  }, {
+    field: 'expectedValue',
+    type: FieldDefinition.Type.ANYSCALAR,
+    description: 'Expected field value',
+  }];
+  protected expectedRecords: ExpectedRecord[] = [{
+    id: 'salesforceObject',
+    type: RecordDefinition.Type.KEYVALUE,
+    fields: [{
+      field: 'Id',
+      type: FieldDefinition.Type.NUMERIC,
+      description: "Object's SalesForce ID",
+    }, {
+      field: 'CreatedDate',
+      type: FieldDefinition.Type.DATETIME,
+      description: "Object's Created Date",
+    }, {
+      field: 'LastModifiedDate',
+      type: FieldDefinition.Type.DATETIME,
+      description: "Object's Last Modified Date",
+    }],
+    dynamicFields: true,
+  }];
+
+  async executeStep(step: Step): Promise<RunStepResponse> {
+    const stepData: any = step.getData().toJavaScript();
+    const field: string = stepData.field;
+    const objName: string = stepData.objName;
+    const id: string = stepData.id;
+    const operator: string = stepData.operator || 'be';
+    const expectedValue: string = stepData.expectedValue;
+    let object: Record<string, any>;
+
+    try {
+      object = await this.client.findObjectById(objName, id);
+    } catch (e) {
+      return this.error('There was a problem checking the Object: %s', [e.toString()]);
```

Comment:
- [ ] Same as above: consider adding the `objName` to make the message friendlier.